src/core_plugins/kibana/public/dashboard/dashboard_state.js

```diff
@@ -128,33 +325,41 @@ export class DashboardState {
    * @param toJson {function} A custom toJson function. Used because the previous code used
    * the angularized toJson version, and it was unclear whether there was a reason not to use
    * JSON.stringify
+   * @param timefilter
    * @returns {Promise<string>} A promise that if resolved, will contain the id of the newly saved
    * dashboard.
    */
-  saveDashboard(toJson) {
+  saveDashboard(toJson, timeFilter) {
```

Comment:
saving the dashboard has so much interaction with appState, so it seems to belong IMO (I will remove reloadDashboardUrl).

Where do you think is a better spot for this?  Or do you think it should be broken out into something like:
```
dashboardState.saveStateToDashboard()
dash.save().then((id) => {
  dashboardState.refreshAppState(id);
}
```
I wanted to move as much code out of the angular directive as possible so it can be re-usable by another UI (react, hehe).  But maybe more, or different, separation would improve readability?  Maybe something like :

```
class SavedDashboard {
  save(dashboardState) {
    dashboardState.saveAppStateToDashboard(this);
    this.save().then((id) => {
      dashboardState.refreshAppState(id);
    }
  }
}
```

But without more refactoring and a larger effort, I don't think I'd gain much benefit from moving some code back into the dashboard directive, so for now I think I'll leave as is.